wolfcrypt/src/evp.c

```diff
@@ -6803,6 +6805,928 @@ void wolfSSL_EVP_PKEY_free(WOLFSSL_EVP_PKEY* key)
+
+            if (i != left - 1) {
+                wsz = 1;
+                if (idx + wsz <= EVP_PKEY_PRINT_LINE_WIDTH_MAX) {
+                    XMEMSET(buff + idx, ':', wsz);
+                    idx += wsz;
+                }
+            }
+        }
+    }
+    if (ret == WOLFSSL_SUCCESS) {
+        ret = wolfSSL_BIO_write(out, buff, idx) > 0;
+    }
+    if (ret == WOLFSSL_SUCCESS) {
+        ret = wolfSSL_BIO_write(out, "\n", 1) > 0;
+    }
+#ifdef WOLFSSL_SMALL_STACK
+    XFREE(buff, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+#endif
+    return ret;
+}
+#if !defined(NO_RSA)
+/* PrintPubKeyRSA is a helper function for wolfSSL_EVP_PKEY_print_public
+ * to parse a DER format RSA public key specified in the second parameter.
+ * Parameters:
+ * out     bio to output dump data
+ * pkey    buffer holding public key data
+ * pkeySz  public key data size
+ * indent  the number of spaces for indent
+ * bitlen  bit size of the given key
+ * pctx    context(not used)
+ * Returns 1 on success, 0 on failure.
+*/
+static int PrintPubKeyRSA(WOLFSSL_BIO* out, const byte* pkey, int pkeySz,
+    int indent, int bitlen, ASN1_PCTX* pctx)
+{
+    byte   buff[8] = { 0 };
+    word32 inOutIdx = 0;
+    word32 nSz;             /* size of modulus */
+    word32 eSz;             /* size of public exponent */
+    const byte*  n   = NULL;
+    const byte*  e   = NULL; /* pointer to modulus/exponent */
+    int idx;
+    int wsz;
+    word32 i;
+    word32 exponent = 0;
+    word32 outSz;
+    byte   outHex[3];
+    mp_int a;
+
+    (void)pctx;
+
+    if( mp_init(&a) != 0) {
+        return WOLFSSL_FAILURE;
+    }
+    if (indent < 0) {
+        indent = 0;
+    }
+    if (indent > EVP_PKEY_PRINT_INDENT_MAX) {
+        indent = EVP_PKEY_PRINT_INDENT_MAX;
+    }
+    /* parse key to get modulus and exponent */
+    if (wc_RsaPublicKeyDecode_ex(pkey, &inOutIdx, pkeySz, 
+                                            &n, &nSz, &e, &eSz) != 0) {
+        return WOLFSSL_FAILURE;
+    }
+
+    /* print out public key elements */
+    idx = 0;
+    XMEMSET(buff, 0, sizeof(buff));
+    Indent(out, indent);
+    if (wolfSSL_BIO_write(out, "RSA Public-Key: (",
+                          (int)XSTRLEN("RSA Public-Key: (")) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+    if (mp_set_int(&a, bitlen) != 0) {
+        return WOLFSSL_FAILURE;
+    }
+    if (mp_todecimal(&a, (char*)buff) != 0) {
+        return WOLFSSL_FAILURE;
+    }
+    wsz = (int)XSTRLEN((const char*)buff);
+
+    if (wolfSSL_BIO_write(out, buff + idx, wsz) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+    if (wolfSSL_BIO_write(out, " bit)\n", (int)XSTRLEN(" bit)\n")) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+    /* print Modulus */
+    Indent(out, indent);
+    if (wolfSSL_BIO_write(out, "Modulus:\n", 
+                                (int)XSTRLEN("Modulus:\n")) <= 0) { 
+        return WOLFSSL_FAILURE;
+    }
+
+    /* print modulus with lenading zero if exists */
+    if (*n & 0x80 && *(n-1) == 0) {
```

Comment:
I can't see n being used after this.
If it has a leading zero than modify n and nSz and then only have one call to DumpElement.

Reply:
Modified as suggested.